client/question/domain/CodeEvalResultObjectFactory.js

```diff
@@ -154,10 +169,7 @@ tie.factory('CodeEvalResultObjectFactory', [
      * @returns {boolean}
      */
     CodeEvalResult.prototype.hasRecursionLimitError = function() {
-      var errorString = this.getErrorString();
-      return errorString && (
-        errorString.startsWith('ExternalError: RangeError') ||
-        errorString.includes('maximum recursion depth exceeded'));
+      return this.getErrorString() && this._memoryLimitExceeded;
```

Comment:
Ditto.

Also maybe rename to hasMemoryLimitError (this is a bit different from recursion limit, which is a subcase). Possibly we should change the error message shown to the user too.

Reply:
I didn't update the error message -- what do you have in mind?

"Your code used more memory than we allow for this exercise; please edit your code to be more space-efficient."?

Reply:
Yes, something like that seems reasonable to me. (Though I might replace the semicolon with a period.)

Thanks!